src/test/java/ArrayMinHeapTest.java

```diff
@@ -37,6 +40,15 @@ void add_empty_returnsTrue() {
                     .isTrue();
         }
 
+        @Test
+        void peek_empty_throwsException() {
+            assertThatExceptionOfType(NoSuchElementException.class)
+                    .isThrownBy(
+                            () -> classUnderTest.peek()
+                    )
```

Comment:
a little overly vertical, imo. `isThrownBy(() -> classUnderTest.peek())` is more ledgible. You can also use a method reference here if you prefer, `isThrownBy(classUnderTest::peek)`

Reply:
niceeee